ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -34,19 +34,32 @@ export type TimeSeriesChartProps = PanelProps<TimeSeriesChartOptions>;
 
 export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
   const {
-    spec: { queries, thresholds },
+    spec: { queries, thresholds, y_axis },
     contentDimensions,
   } = props;
 
   // popuate default 'position' and other future properties
   const legend = props.spec.legend ? merge({}, DEFAULT_LEGEND, props.spec.legend) : undefined;
 
-  const unit = props.spec.unit ?? DEFAULT_UNIT;
+  // TODO: eventually remove props.spec.unit, add support for y_axis_alt.unit
+  let unit = DEFAULT_UNIT;
+  if (props.spec.y_axis?.unit) {
+    unit = props.spec.y_axis.unit;
```

Comment:
@saminzadeh good points about unit / how the tooltip should work in the future, I ended up adding `unit` inside y_axis here. Currently this will override the top level spec.unit, but eventually I'll probably get rid of spec.unit completely